Dockerfile

```diff
@@ -0,0 +1,43 @@
+#!/bin/echo docker build . -f
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: MPL-2.0
+#{
+# Copyright: 2021+ Philippe Coval <https://purl.org/rzr/>
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/ .
+#}
+
+FROM rust:bullseye
+LABEL maintainer="Philippe Coval (philippe.coval@astrolabe.coop)"
+
+RUN echo "#log: Setup system" \
+  && set -x \
+  && apt-get update \
+  && apt-get install -y \
+      --no-install-recommends \
+     ca-certificates \
+     libssl-dev \
+     pkg-config \
+  && apt-get upgrade -y \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/* \
+  && date -u
+
+ARG project=astarte-device-sdk-rust
+ARG workdir="/local/src/${project}"
+
+COPY . "${workdir}/"
+WORKDIR "${workdir}"
+RUN echo "#log: ${project}: Building sources" \
+  && set -x \
+  && export OPENSSL_NO_VENDOR=1 \
+  && cargo build  \
+  && cargo install --path . \
+  || cargo install --examples --path . \
+  && date -u
+
+WORKDIR "${workdir}"
+ENTRYPOINT [ "cargo" ]
+CMD [ "run", "--example", "simple", "--help" ]
```

Comment:
I'm not sure what the expected usage is, feel free to add instructions into the readme

Reply:
this is the default run command , it could be overridden when using docker run ...

Reply:
Yeah but now it's calling --help for cargo, overriding --example, it doesn't do anything as it is. I can get what the idea was but I think we should fix this and document the intended usage anyway, because a dockerfile can be use in different ways (for example to produce binaries in a volume)

Reply:
--help for example will be displayed not cargo